### DISCUSSION TITLE
Issue in @types/office-js

### DISCUSSION THREAD

**Discussion by sweco-seabth** [DefinitelyTyped/DefinitelyTyped#65852] [Issues with a @types package]
node_modules/@types/office-js/index.d.ts' is not a module. I am getting this issue when i try to use this.

[a comment by an automated account is not shown]

**Comment by ElizabethSamuel-MSFT** [comment]
@sweco-seabth Can you provide more information about where you're trying to use this type?

For an example of how you could access the Office JavaScript API and types, see [Enabling IntelliSense for a TypeScript project](https://learn.microsoft.com/office/dev/add-ins/develop/referencing-the-javascript-api-for-office-library-from-its-cdn#enabling-intellisense-for-a-typescript-project).

**Reply by sweco-seabth** [reply]
I am trying to import here "import * as Office from "office-js";"  here is the plac i am getting error. Actually I am trying to add some contents inside my powerpoint controls using the Addins. So I  tried to import the office -js in my typescript file.

**Reply by ElizabethSamuel-MSFT** [reply]
Thanks for your response. Did the information at the link I shared help get you unblocked?

If not, do provide any other info you can for debugging e.g., framework, IDE, sample code.

Thanks.

**Reply by sweco-seabth** [reply]
Thanks for your response. "npm install --save-dev @types/office-js" this thing doesn't help me. I am trying to push some of contents in to the powerpoint slide controls that's where I am trying to import this office-js. Also I read an discussions from here https://github.com/microsoft/TypeScript/issues/11420. Saying like this "The declaration file does not say that there is a module called office-js. it just says there is a global variable called Office. using Office.select for instance is allowed, but import .. from "office-js" is not.

please see http://www.typescriptlang.org/docs/handbook/modules.html for more details about modules." is that true. Am I trying to do something wrong?

**Reply by ElizabethSamuel-MSFT** [reply]
Maybe the following can help you.

- [Referencing the Office JavaScript API library](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/referencing-the-javascript-api-for-office-library-from-its-cdn)
- [Build your first PowerPoint task pane add-in](https://learn.microsoft.com/en-us/office/dev/add-ins/quickstarts/powerpoint-quickstart?tabs=yeomangenerator) (quick start): If you try the yeoman generator option, then consider choosing the TypeScript script type instead of JavaScript when prompted.
- [Basic API call (TypeScript)](https://github.com/OfficeDev/office-js-snippets/blob/main/samples/powerpoint/basics/basic-api-call-ts.yaml) PowerPoint code sample: You can try this out in [Script Lab](https://learn.microsoft.com/en-us/office/dev/add-ins/overview/explore-with-script-lab).

If you can share some more details about your setup, perhaps we can provide more pointed guidance.

**Comment by ElizabethSamuel-MSFT** [comment]
@sweco-seabth Are you still hitting this issue?

We've since updated items tagged as "module" in the office-js and office-js-preview files so they're now "namespace".

**Reply by ElizabethSamuel-MSFT** [reply]
@sweco-seabth Also, I'm not sure if we made this clear at the time but @types/office-js is for the IntelliSense. When you're using the Office.js API in a TypeScript project, the reference to these types will provide the documentation for the APIs. If you want to use the Office.js APIs, the usual way is to add the appropriate `<script>` tag to the `<head>` of your relevant HTML files, although the way you reference the API library may depend on the framework you're using. See [Referencing the Office JavaScript API library](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/referencing-the-javascript-api-for-office-library-from-its-cdn) for details.
